frontend/src/components/utils/DatamapsExtension.tsx

```diff
@@ -33,8 +29,15 @@ export default class Datamap extends React.Component<Props> {
     this.drawMap();
   }
 
+  shouldComponentUpdate(nextProps: Props) {
+    if (nextProps.pins.length !== this.props.pins.length) {
```

Comment:
Let's do deep comparison here just to be on the safe side of things. Use https://www.npmjs.com/package/fast-deep-equal (we already depend on it through other dependencies, so it will not introduce extra code)